tests/unit/tty.test.ts

```diff
@@ -0,0 +1,18 @@
+import { isatty } from "tty";
+
+describe("tty", () => {
+  it("isatty", () => {
+    // FIXME: in the test environment, isatty is always false
+    expect([
+      isatty(1),
+      isatty(2),
+      isatty(3),
+      isatty(4),
+    ]).toEqual([
+      false,
+      false,
+      false,
+      false,
+    ]);
+  });
```

Comment:
Do I get the same results if I run it on the laptop at hand?

Reply:
The results of running the command line are correct. I suspect that llrt test uses redirection or other unknown behaviors, which causes all false to be returned in the test.

```
import { isatty } from 'tty'

console.log(
  isatty(1),
  isatty(2),
  isatty(3),
  isatty(4),
)
```

```
llrt ./a.js

true true false false
```

Reply:
I figured that would probably be the case. This is a little troubling because I always run the test suite on the laptop at hand to make sure it is error free before shipping...

Reply:
Now only the return value type is checked

Reply:
I think a way to properly test this is to add a rust test for this (we can skip js test) where we launch a child process to dermine if file descriptor is tty. This has the be adjusted for posix and windows

Reply:
Can you explain it in more detail? The difficulty in testing is that isatty always returns false in github action
https://github.com/ahaoboy/tty-test/actions/runs/11886599988/job/33118007342

Reply:
The easiest way I've found to test this is to distinguish whether the program is running in GitHub action or not.

Reply:
You can run "test -t" from a rust child process:
```rust
let status = Command::new("test")
        .arg("-t")
        .arg("1") // File descriptor 1 is stdout,
        .status(); 
```